HeatApp/components/screens/UserSettingsScreen.js

```diff
@@ -14,27 +16,202 @@ export default class UserSettingsScreen extends React.Component {
+    3: 'Other'
+  };
+
   constructor(props) {
     super(props);
+    this.state = {
+      gender: 3,
+    };
+  }
+
+  getStateClone() {
+    var stateClone = {};
+    Object.keys(this.state).forEach(key => {
+      stateClone[key] = this.state[key];
+    });
+    return stateClone;
+  }
+
+  updateState(field, value) {
+    var newState = this.getStateClone();
+    newState[field] = value;
+    this.setState(newState);
+  }
+
+  updateUsername(text) {
+    this.updateState('username', text);
+  }
+
+  updatePassword(text) {
+    this.updateState('password', text);
+  }
+
+  updateWeight(text) {
+    this.updateState('weight', parseFloat(text));
+  }
+
+  updateHeight(text) {
+    this.updateState('height', parseFloat(text));
+  }
+
+  updatePhysicalActivity(text) {
+    this.updateState('physicalActivity', text);
+  }
+
+  updateTargetWeight(text) {
+    this.updateState('targetWeight', parseFloat(text));
+  }
+
+  updateGender(value) {
+    this.updateState('gender', value);
+  }
+
+  updateBirthdate(value) {
+    this.updateState('birthdate', value);
+  }
+
+  genderSelection() {
+    this.updateState('showGenderSelection', !this.state.showGenderSelection);
+  }
+
+  birthdateSelection() {
+    if (!this.state.showBirthdateSelection && !this.state.birthdate) {
+      var newState = this.getStateClone();
+      newState['showBirthdateSelection'] = true;
+      newState['birthdate'] = new Date();
+      this.setState(newState);
+    } else {
+      this.updateState('showBirthdateSelection', !this.state.showBirthdateSelection);
+    }
   }
 
   render() {
+    // TODO: Add animation for pickers.
     return (
-      <View style={[styles.container]}>
-        <Text style={styles.text}>User settings screen</Text>
-      </View>
+      <ScrollView contentContainerStyle={styles.contentContainer}>
+        <View style={[styles.container]}>
+          <SettingsTextInput hintText="Username"
+                             placeholder="Not set"
+                             text={this.state.username}
+                             textUpdate={this.updateUsername.bind(this)}/>
+          <SettingsTextInput hintText="Password"
+                             placeholder="Password"
+                             secureTextEntry={true}
+                             text={this.state.password}
+                             textUpdate={this.updatePassword.bind(this)}/>
+
+          <Divider style={styles.divider}/>
+
+          <TouchableOpacity onPress={this.genderSelection.bind(this)}>
+            <View style={styles.inputWrap}>
+              <View style={styles.hintTextWrapper}>
+                <Text style={styles.textHint}>Gender</Text>
+                <Text style={styles.textValue}>
+                  {this.genderMatch[this.state.gender]}
+                </Text>
+              </View>
+            </View>
+          </TouchableOpacity>
+
+          {!!this.state.showGenderSelection &&
+            <PickerIOS selectedValue={this.state.gender}
```

Comment:
It's better to add check of platform used, because on Android phones it returns error.